### PR TITLE
Add have_downloaded matcher

### PR DESCRIPTION
**Pull request by damselem**

### CODE REVIEW

File: lib/capistrano/spec.rb [hunk 1]
```diff
@@ -182,6 +190,30 @@ def to(to)
...
+          downloads = downloads.select { |f, d| d[:from] == @from } if @from
+          downloads.any?
+        end
+
+        def from(from)
+          @from = from
+          self
+        end
+
+        failure_message_for_should do |actual|
```
Comment (jhbabon) [inline lib/capistrano/spec.rb:208]: Maybe we can add a `failure_message_for_should_not` block too.

File: spec/capistrano-spec_spec.rb [hunk 2]
```diff
@@ -40,6 +40,19 @@
     end
   end
 
+  describe 'have_downloaded' do
+
+    it "will not raise error when download is in recipe" do
+      fake_recipe.find_and_execute_task('fake:thing')
+      expect{ should have_downloaded('foo').from('/tmp/foo')}.to_not raise_error(RSpec::Expectations::ExpectationNotMetError, /expected configuration to download .*\s* from .*\s* but did not/)
```
Comment (jhbabon) [inline spec/capistrano-spec_spec.rb]: I know that there other lines very long in the spec, but maybe we can start making the lines more short. Maybe we can follow some style guide like this[1]

[1] https://github.com/bbatsov/ruby-style-guide

File: spec/downloaded_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        download 'source.file', 'temp.XC3PO.file' # E.g. downloading to temp file
+      end
+    end
+  end
+
+  subject(:configuration) { @configuration }
+
+  it 'some file' do
+    configuration.download 'source.file', 'target.file'
+    expect{ should have_downloaded }
```
Comment (jhbabon) [inline spec/downloaded_spec.rb]: I'm not 100% sure here, but if you use `expect`, you don't need to use `should`. Maybe these specs can be written like this:

``` ruby
# using expect
expect { subject.download 'source.file', 'target.file' }.to have_downloaded

# using should
subject.download 'source.file', 'target.file'
subject.should have_downloaded
```

This post explains more about `expect` and `should`[1]

[1] http://myronmars.to/n/dev-blog/2012/06/rspecs-new-expectation-syntax

Comment (damselem) [inline spec/downloaded_spec.rb]: Makes total sense but I wanted to stick with the style used in the other specs. Should I make changes to the other ones as well?

Comment (jhbabon) [inline spec/downloaded_spec.rb]: I think the case is different. In the other specs we want to test two things:
- That the behaviour is met.
- That no error is raised.

Here you only want to test the behaviour so maybe the use of one syntax is enough.

**Comment by damselem** [comment]
Is there anything else required to merge this?

**Comment by jhbabon** [comment]
Maybe you can check this commit: https://github.com/technicalpickles/capistrano-spec/pull/16#discussion_r5470018

But, for the rest, is good for me. I don't know what thinks the rest of people. @technicalpickles, @relistan, @petems ?

**Comment by petems** [comment]
Looks good to me tbh, I'm down to merge now.

**Comment by jhbabon** [comment]
@damselem sorry for the delay with this. You already have the new matcher in the version `0.6.0`. Thanks for your collaboration :smile: :+1:

**Comment by damselem** [comment]
Np, thanks to you for supporting this library!

**Pull request merged**